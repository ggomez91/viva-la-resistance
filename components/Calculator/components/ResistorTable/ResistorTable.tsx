import React from "react";
import styles from "./ResistorTable.module.scss";

type ResistorTableProps = {
  ohms: number;
  tolerancePercent: number;
};

/**
 * Renders a table showing the calculated values of the resistor
 */
const ResistorTable: React.FC<ResistorTableProps> = ({
  ohms,
  tolerancePercent,
}) => {
  return (
    <div className={styles.ResistorTable}>
      <table className="table">
        <thead>
          <tr>
            <th scope="col">Resistance</th>
            <th scope="col">Tolerance</th>
            <th scope="col">Minimum</th>
            <th scope="col">Maximum</th>
          </tr>
        </thead>
        <tbody>
          <tr>
            <td>{ohms} ohms</td>
            <td>Â± {tolerancePercent} % </td>
            <td data-testid="min">
              {ohms - (tolerancePercent * ohms) / 100} ohms
            </td>
            <td data-testid="max">
              {ohms + (tolerancePercent * ohms) / 100} ohms
            </td>
          </tr>
        </tbody>
      </table>
    </div>
  );
};

export default ResistorTable;
