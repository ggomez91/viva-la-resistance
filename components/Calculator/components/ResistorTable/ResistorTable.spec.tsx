import React from "react";
import renderer from "react-test-renderer";
import ResistorTable from "./ResistorTable";
import Enzyme, { mount } from "enzyme";

import Adapter from "@wojtekmaj/enzyme-adapter-react-17";
Enzyme.configure({ adapter: new Adapter() });

describe("ResistorTable", () => {
  it("renders correctly", () => {
    const tree = renderer
      .create(<ResistorTable ohms={50} tolerancePercent={34} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("Calculates the minimum and maximum values", () => {
    const wrapper = mount(<ResistorTable ohms={100} tolerancePercent={1} />);
    expect(wrapper.find('[data-testid="max"]').text()).toBe("101 ohms");
    expect(wrapper.find('[data-testid="min"]').text()).toBe("99 ohms");
  });
});
