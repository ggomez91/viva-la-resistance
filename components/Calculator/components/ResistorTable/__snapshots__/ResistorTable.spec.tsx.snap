// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ResistorTable renders correctly 1`] = `
<div>
  <table
    className="table"
  >
    <thead>
      <tr>
        <th
          scope="col"
        >
          Resistance
        </th>
        <th
          scope="col"
        >
          Tolerance
        </th>
        <th
          scope="col"
        >
          Minimum
        </th>
        <th
          scope="col"
        >
          Maximum
        </th>
      </tr>
    </thead>
    <tbody>
      <tr>
        <td>
          50
           ohms
        </td>
        <td>
          Â± 
          34
           % 
        </td>
        <td
          data-testid="min"
        >
          33
           ohms
        </td>
        <td
          data-testid="max"
        >
          67
           ohms
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;
